### PR TITLE
[Arm64/Ubuntu] CI match arm32 except triggers

### PR DESCRIPTION
**Pull request by sdmaclea**
@BruceForstall @jashook I tried to make arm64 match arm32 as much as possible with this patch.  PTAL as this will like need aesthetic changes at a minimum.

cc/ @debayang @dotnet/arm64-contrib

### CODE REVIEW

File: netci.groovy [hunk 1]
```diff
@@ -2486,7 +2486,7 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
                     // ZIP up the generated CoreFX runtime and tests.
 
                     def dockerImage = getDockerImageName(architecture, os, true)
-                    def dockerCmd = "docker run -i --rm -v \${WORKSPACE}:\${WORKSPACE} -w \${WORKSPACE} -e ROOTFS_DIR=/crossrootfs/arm -e CAC_ROOTFS_DIR=/crossrootfs/x86 ${dockerImage} "
+                    def dockerCmd = "docker run -i --rm -v \${WORKSPACE}:\${WORKSPACE} -w \${WORKSPACE} -e ROOTFS_DIR=/crossrootfs/${architecture} -e CAC_ROOTFS_DIR=/crossrootfs/x86 ${dockerImage} "
```
Comment (jashook) [inline netci.groovy]: cac_rootfs should be x64? However, should remain x86 for arm.

**Comment by jashook** [comment]
I believe it should, because corefx arm64 packages exist now; however, it is worth confirming.

**Comment by jashook** [comment]
Thanks for the work!

File: netci.groovy (hunk 1 above)
Comment (echesakovMSFT) [inline netci.groovy]: Just wondering if setting `CAC_ROOTFS_DIR` to some non-empty string value for `arm64` gonna break build?
In official build it is set to empty string for `arm64` https://github.com/dotnet/coreclr/blob/359f32036f0e72280c37343f7c3af1bbe0582828/buildpipeline/DotNet-CoreClr-Trusted-Linux-Crossbuild.json#L527-L530.

Comment (sdmaclea) [reply netci.groovy]: I was hoping it would be ignored.  The alternative would be to special case it.

Comment (sdmaclea) [reply netci.groovy]: It won't work

Comment (sdmaclea) [reply netci.groovy]: Fixed

File: netci.groovy [hunk 2]
```diff
@@ -323,14 +323,6 @@ class Constants {
...
     ]
   
-    def static validLinuxArm64Scenarios = [ 
-               'innerloop',
-               'normal',
-               'r2r',
-               'gcstress0x3',
-               'gcstress0xc'
-    ]
-
```
Comment (sdmaclea) [inline netci.groovy]: I wonder if I should keep a separate arm64 list.  gcServer may be interesting on arm64 while not on arm.

**Comment by sdmaclea** [comment]
> It is worth verifying that the following works for arm64 linux:

>> buildCommands += "${dockerCmd}${WORKSPACE}/build-test.sh ${lowerConfiguration} ${architecture} cross generatelayoutonly"

It is working on the tip

**Comment by sdmaclea** [comment]
@BruceForstall ping

When it is time someone should trigger "test ci" (I do not have permissions)

File: netci.groovy [hunk 3]
```diff
@@ -620,9 +612,14 @@ def static setMachineAffinity(def job, def os, def architecture, def options = n
         assert os in supportedArmLinuxOs
 
         if (architecture == 'arm64') {
-            if ((options != null) && (options['is_build_only'] == true)) {
-                // Arm64 Linux build machine
-                Utilities.setMachineAffinity(job, os, 'arm64-cross-latest')
+            assert (architecture == 'arm64') && (os == 'Ubuntu')
+            def isFlow  = (options != null) && (options['is_flow_job'] == true)
+            def isBuild = (options != null) && (options['is_build_job'] == true)
```
Comment (BruceForstall) [inline netci.groovy]: setJobAffinity uses `affinityOptions = ['is_build_only': true]`. You could probably change it to use `is_build_job` instead, in which case this should work.

**Comment by sdmaclea** [comment]
test OSX10.12 x64 Checked Innerloop Build and Test
test Windows_NT arm64 Cross Checked Innerloop Build and Test

**Comment by sdmaclea** [comment]
@BruceForstall I believe I addressed your comments

**Comment by BruceForstall** [comment]
@sdmaclea I'll re-review soon (I've been out)

File: netci.groovy [hunk 4]
```diff
@@ -1021,6 +1025,11 @@ def static getDockerImageName(def architecture, def os, def isBuild) {
                 return "microsoft/dotnet-buildtools-prereqs:ubuntu-14.04-cross-e435274-20180426002420"
             }
         }
+        else if (architecture == 'arm64') {
+            if (os == 'Ubuntu') {
+                return "microsoft/dotnet-buildtools-prereqs:ubuntu-16.04-cross-arm64-a3ae44b-20180315221921
```
Comment (BruceForstall) [inline netci.groovy]: missing a trailing quote:
```
13:27:02 ERROR: startup failed:
13:27:02 netci.groovy: 1030: expecting anything but ''\n''; got it anyway @ line 1030, column 108.
13:27:02    s-arm64-a3ae44b-20180315221921
13:27:02                                  ^
13:27:02 
13:27:02 1 error
```

Comment (sdmaclea) [reply netci.groovy]: Done

**Comment by BruceForstall** [comment]
@dotnet-bot test ci please

**Comment by sdmaclea** [comment]
```
14:07:37 netci.groovy: 3124: expecting ')', found 'def' @ line 3124, column 5.
14:07:37        def doCoreFxTesting = isCoreFxScenario(scenario)
```

**Comment by BruceForstall** [comment]
@dotnet-bot test ci please

**Comment by BruceForstall** [comment]
@dotnet-bot test ci please

**Comment by sdmaclea** [comment]
> Have you tested building using Docker manually?

Official builds have been using docker to build since March.  I tested manually when I created the docker image for that.

> So, can you add some code at the top of addNonPRTriggers to disable all arm64 cron jobs?

Sure

**Comment by BruceForstall** [comment]
btw, here is the base/diff comparison of jobs before/after this change:

[diff.txt](https://github.com/dotnet/coreclr/files/2014029/diff.txt)
[base.txt](https://github.com/dotnet/coreclr/files/2014030/base.txt)

**Comment by sdmaclea** [comment]
Added condition to addNonPRTriggers and squashed.

**Comment by BruceForstall** [comment]
@dotnet-bot test ci please

**Pull request merged**